provider/negotiate.go

```diff
@@ -6,10 +6,11 @@ import (
 	"skybin/core"
 )
 
+// Helper to negotiate contracts
```

Comment:
I think it should be concurrency safe just from SQLite (see part 5 of their FAQ). As I understand, it locks the DB for every write op, but reads can occur simultaneously. Please correct me if I'm wrong, I tried to read up on it, but I'm struggling to understand where and when to lock these concurrent operations.

"Multiple processes can have the same database open at the same time. Multiple processes can be doing a SELECT at the same time. But only one process can be making changes to the database at any moment in time, however."

Reply:
Right. My concern is around the in-memory fields, like the provider.renters map, the renter info objects, and the provider stats fields. Addressing these should hopefully be pretty simple. First add a sync.Mutex to the provider struct (it may be in there). Then lock around updates to the in-memory fields when they are updated or read in a request handler. Request handlers can run at the same time, including two concurrent executions of the same handler for different requests.